src/lib/deploy.ts

```diff
@@ -148,7 +152,32 @@ export async function deployImpl(this: Deploy, cl: typeof Deploy) {
     const previouslyDeployedDeal =
       workersConfig.deals?.[fluenceEnv]?.[workerName];
 
+    let isDealUpdate = previouslyDeployedDeal !== undefined;
+
     if (previouslyDeployedDeal !== undefined) {
+      const deal = readonlyDealClient.getDeal(
+        previouslyDeployedDeal.dealIdOriginal,
+      );
+
+      const status = await deal.getStatus();
+      const isDealEnded = status === 2n;
```

Comment:
maybe convert it first and compare with a string? we sometimes change enum order, and it will break this code without anyone knowing...

Reply:
Yeah totally forgot to adjust that, ideally I would use enum exposed by deal-ts-client lib, but I think it's not exposed

Reply:
it is exposed, I believe https://github.com/fluencelabs/deal/blob/117d49d9f9bb0dfb7e4b5efbff7fc03db1deeec5/ts-client/src/dealExplorerClient/types/schemes.ts#L282

Reply:
The link leads to some type, but I need actual values, e.g. an enum